### ISSUE TITLE
ADD fields in llx_product and llx_product_batch

### ISSUE THREAD

**Issue by lmag** [Dolibarr/dolibarr#16992]
# Feature Request
We are working on the equipment module as such we would like to add fields

1) Add in
https://wiki.dolibarr.org/index.php?title=Table_llx_product

"lifetime";type:Date;Sample:30/06/2021
"control_periodicity";Type:Période;Samples:365d,12m,1y

2) Add in
https://wiki.dolibarr.org/index.php?title=Table_llx_product_batch

"lifetime";type:Date;Sample:30/06/2021
"date_of_manufacture";type:Date;Sample:30/06/2021
"date_of_destruction";type:Date;Sample:30/06/2021
"date_of_commissioning";type:Date;Sample:30/06/2021
"control_periodicity";Type:Période;Samples:365d,12m,1y
"Statut" >> Dictionnaire:Value(OK;KO)

## Suggested steps
[*List of tasks to achieve goal*]We would like to have a feedback in order to be sure to do the right PR

**Comment by lmag** [comment]
@altatof @atm-florian 
Voici la proposition à discuter avec @eldy pour le modèle de données

**Comment by BebZ** [comment]
Hello, very nice to have interest on this module :) 
- do you use it to make it "equipment-module-compatible" ? 
- or just to increase the number of fields that can be populated for product's batches ?
(if second option I guess the answer would be there are extrafields also for batch/dluo module)

Do you have a specific goal to achieve ? (ISO 13485 ? MDR 2017/745 ? GLP compliance ? )

Thanks again !

**Comment by lmag** [comment]
ISO 13485: 2016 concerns the medical environment as well as the other references that you have proposed but it can be a good idea.
In the idea, we want to complete the data diagram with the information common to a product in order to be able to manage its states and edit the life sheets of batches or products.

We could very well use the extrafields but we believe that these fields are missing in dolibarr. We believe that an equipment module should be present in an ERP.

**Comment by JoAnMaFe** [comment]
Some time ago, I requested something similar. 
Additional technical data sheet for products # 15822

**Comment by BebZ** [comment]
I think you can find the extrafields here, it should not be missing : 
/product/admin/product_lot_extrafields.php?action=create#newattrib

**Comment by lmag** [comment]
@eldy do you have any advice before we start PR

**Comment by homer8173** [comment]
It seems that the develop branch is affected : 
Erreur : Unknown column 'lifetime' in 'field list' - UPDATE llx_product SET label = 'testProd1', ref = 'testProd1', ref_ext = null, default_vat_code = null, tva_tx = 20, recuperableonly = 0, localtax1_tx = 0, localtax2_tx = 0, localtax1_type = '0', localtax2_type = '0', barcode = null, fk_barcode_type = 2, tosell = 1, tobuy = 1, tobatch = 0, batch_mask = '', finished = null, net_measure = null, net_measure_units = null, weight = null, weight_units = '0', length = null, length_units = '0', width= null, width_units = '0', height = null, height_units = '0', surface = null, surface_units = '0', volume = null, volume_units = '0', fk_default_warehouse = null, seuil_stock_alerte = 0, description = '', url = null, customcode = '', fk_country = null, fk_state = null, lifetime = null, qc_frequency = null, note = '', duration = '', accountancy_code_buy = '', accountancy_code_buy_intra = '', accountancy_code_buy_export = '', accountancy_code_sell= '', accountancy_code_sell_intra= '', accountancy_code_sell_export= '', desiredstock = 0, cost_price = null, fk_unit= NULL, price_autogen = 0, fk_price_expression = NULL, fk_user_modif = 1 WHERE rowid = 2
